rand_core/src/impls.rs

```diff
@@ -160,6 +160,49 @@ pub fn next_u64_via_fill<R: RngCore + ?Sized>(rng: &mut R) -> u64 {
     u64::from_le_bytes(buf)
 }
 
+/// Implement [`TryRngCore`] for a type implementing [`RngCore`].
+///
+/// Ideally, `rand_core` would define blanket impls for this, but they conflict with blanket impls
+/// for `&mut R` and `Box<R>`, so until specialziation is stabilized, implementer crates
+/// have to implement `TryRngCore` directly.
+#[macro_export]
+macro_rules! impl_try_rng_from_rng_core {
+    ($t:ty) => {
+        impl $crate::TryRngCore for $t {
+            type Error = core::convert::Infallible;
+
+            #[inline]
+            fn try_next_u32(&mut self) -> Result<u32, Self::Error> {
+                Ok(self.next_u32())
+            }
+
+            #[inline]
+            fn try_next_u64(&mut self) -> Result<u64, Self::Error> {
+                Ok(self.next_u64())
+            }
+
+            #[inline]
+            fn try_fill_bytes(&mut self, dst: &mut [u8]) -> Result<(), Self::Error> {
+                self.fill_bytes(dst);
+                Ok(())
+            }
+        }
+    };
+}
```

Comment:
Caveat: the macro name makes this look like it requires that `$t: RngCore`, but it doesn't actually; it just uses the methods, assuming that `RngCore` is in scope.

Suggestion: use `Ok($crate::RngCore::next_u32(self))` instead so that (1) this *does* require `$t: RngCore` and (2) it does not require that `RngCore` is in scope.

Reply:
Fixed.